common/tests/unittests/test_model_filter.py

```diff
@@ -0,0 +1,40 @@
+import pytest
+
+from autogluon.common.model_filter import ModelFilter
+
+
+@pytest.mark.parametrize(
+    "models,included_model_types,excluded_model_types,expected_answer",
+    [
+        ({"dummy": {}}, ["dummy"], None, {"dummy": {}}),
+        ({"dummy": {"dummy": 1}}, ["dummy"], None, {"dummy": {"dummy": 1}}),
+        ({"dummy": {}}, ["foo"], None, {}),
+        ({"dummy": {}, "foo": {}}, ["foo"], None, {"foo": {}}),
+        ({}, ["foo"], None, {}),
+        ({"dummy": {}}, None, ["dummy"], {}),
+        ({"dummy": {"dummy": 1}}, None, ["dummy"], {}),
+        ({"dummy": {}}, None, ["foo"], {"dummy": {}}),
+        ({"dummy": {}, "foo": {}}, None, ["foo"], {"dummy": {}}),
+        ({}, None, ["foo"], {}),
+        ({"dummy": {}}, ["dummy"], ["dummy"], {"dummy": {}}),
+        ({"dummy": {"dummy": 1}}, ["dummy"], ["dummy"], {"dummy": {"dummy": 1}}),
+        ({"dummy": {}}, ["foo"], ["dummy"], {}),
+        ({"dummy": {}, "foo": {}}, ["foo"], ["dummy"], {"foo": {}}),
+        ({}, ["foo"], ["foo"], {}),
+        (["dummy"], ["dummy"], None, ["dummy"]),
+        (["dummy"], ["foo"], None, []),
+        (["dummy", "foo"], ["foo"], None, ["foo"]),
+        ([], ["foo"], None, []),
+        (["dummy"], None, ["foo"], ["dummy"]),
+        ([], None, ["foo"], []),
+        (["dummy"], None, ["dummy"], []),
+        (["dummy", "foo"], None, ["dummy"], ["foo"]),
+    ],
+)
```

Comment:
It may be easier to have 1 line per argument for each test with proper use of whitespace, otherwise it is very hard to verify correctness by simply looking at this block.